prov/gni/include/gnix_wait.h

```diff
@@ -68,6 +68,8 @@ struct gnix_fid_wait {
 	struct slist set;
 };
 
+extern struct dlist_entry gnix_nic_list;
```

Comment:
Oh I see.  I'd rather this declaration be moved in to gnix_nic.h and then have gnix_wait.h (or gnix_wait.c) include gnix_nic.h